### PR TITLE
Swift storage backend

### PR DESCRIPTION
**Pull request by buchanae**
Work in progress

### CODE REVIEW

**Comment by buchanae** [comment]
Needs tests, but otherwise seems to work in my basic tests on exastack.

Need to figure out how to test something like this (and Google Storage, AWS S3, etc).

**Comment by buchanae** [comment]
zomg, it builds.

**Comment by buchanae** [comment]
The swift backend was not failing when it failed to access files owned by root. So there's a bug in there somewhere.

**Comment by adamstruck** [comment]
Can you also add a test for get/put on a directory?

File: config/config.go [hunk 1]
```diff
@@ -255,6 +258,22 @@ func (l S3Storage) Valid() bool {
...
 
+// SwiftStorage configures OpenStack Swift object storage.
+type SwiftStorage struct {
+	UserName   string
+	Password   string
+	AuthURL    string
+	TenantName string
+	TenantID   string
+	RegionName string
+}
```
Comment (adamstruck) [inline config/config.go]: Could you add an option for these to be loaded from environmental variables?

**Comment by adamstruck** [comment]
- [ ] load swift config vars from environment
- [ ] squash commits
- [ ] rebase

**Pull request merged**